src/quo2/components/graph/wallet_graph/view.cljs

```diff
@@ -0,0 +1,66 @@
+(ns quo2.components.graph.wallet-graph.view
+  (:require [quo2.theme :as theme]
+            [react-native.linear-gradient :as linear-gradient]
+            [react-native.charts :as charts]
+            [react-native.core :as rn]
+            [quo2.components.graph.wallet-graph.style :as style]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.graph.wallet-graph.utils :as utils]))
+
+(defn- wallet-graph-internal
+  [{:keys [data state time-frame theme]}]
+  (let [max-data-points (case time-frame
+                          :empty    0
+                          :1-week   7
+                          :1-month  30
+                          :3-months 90
+                          :1-year   365
+                          500)
+        data            (if (and (not= time-frame :empty) (> (count data) max-data-points))
+                          (utils/downsample-data data max-data-points)
+                          data)
+        max-value       (when-not (= time-frame :empty) (utils/find-highest-value data))
+        width           (:width (rn/get-screen))
```

Comment:
Can we use the re-frame sub `:dimensions/window-width`? 
I believe that the `width` will be the same for both the `screen` and `windows` dimensions for both platforms. Only the `height` varies on the `screen` and `windows` dimensions for Android.

Reply:
I think we should not use subscriptions in quo2 components, that's why I didn't use that sub. I could use `rn/get-window` instead, or receive window-width as a parameter, but as we can get window width from React Native default APIs, I think using `rn/get-window` would be fine. Does that make sense to you?

Reply:
Correct, we shouldn't. For the record, this is in the guidelines.

> Components inside `src/quo2/` should not rely on re-frame, i.e. they should not
dispatch events or use subscriptions.